SU2_CFD/src/variables/CNEMOEulerVariable.cpp

```diff
@@ -187,6 +187,8 @@ bool CNEMOEulerVariable::Cons2PrimVar(su2double *U, su2double *V,
   Tmin   = 50.0; Tmax   = 8E4;
   Tvemin = 50.0; Tvemax = 8E4;
 
+  if (U[nSpecies+nDim+1] <= 0) {U[nSpecies+nDim+1] = 0;}
```

Comment:
how often does this occur? 

I believe we have a check nonphys function

Reply:
In my case, for (N,N2) cold flows it would be a problem because the vibrational energy would return negative due to numerical errors (as in the value would be ~-1E-16). When this happened, M++ would return NaN Temperature, which crashed the simulation. 

That filter would be to prevent the negative values of vibrational energy